File: 0214-shortest-palindrome/0214-shortest-palindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var shortestPalindrome = function(s) {
    let reversed = s.split('').reverse().join('');
    let n = s.length;
    for(let i=0;i<n;i++){
        if(s.substring(0,n-i) === reversed.substring(i)){
            return reversed.substring(0,i)+s;
        }
    }
    return s;
    
};

File: 0214-shortest-palindrome/0214-shortest-palindrome.java
class Solution {
    public String shortestPalindrome(String s) {
        int n = s.length();
        String reversed = new StringBuilder(s).reverse().toString();
        for(int i=0;i<n;i++){
            if(s.substring(0,n-i).equals(reversed.substring(i))){
                return new StringBuilder(reversed.substring(0,i)).append(s).toString();
            }
        }
        return "";
    }
}